### ISSUE TITLE
bug: no-initialize-state false positive when state is already initialized

### ISSUE THREAD

**Issue by agawley** [NickvanDyke/eslint-plugin-react-you-might-not-need-an-effect#48]
### Did you check docs and existing issues?

- [x] I have read the plugin docs
- [x] I have searched the existing issues
- [x] I have read [You Might Not Need An Effect](https://react.dev/learn/you-might-not-need-an-effect) and am reasonably confident that I *do* need an effect

### Plugin version

0.5.6

### ESLint version

9.33.0

### Describe the bug

```
export const useSomeStateThatChangesOnFirstRender = () => {
    const [state, setState] = useState(false);
    useEffect(() => setState(true), []);
    return state;
};
```
This code returns a warning on the setState call
`Avoid initializing state in an effect. Instead, pass "state"'s initial value to its useState .eslint(react-you-might-not-need-an-effect/no-initialize-state)`

The state is not being initilised in the effect. The code already does what the warning suggests.

### Steps To Reproduce

1. Type the above code into a file
2. Run eslint on that file

### Expected Behavior

Should not trigger this error (or the error should be reworded)

**Comment by NickvanDyke** [comment]
Thanks for the report - do you have a more real-world example of why we'd want to initialize the state to one value, and then immediately set it to another value in an effect? Rather than just initialize it to the second value. Or you're just offering input on the error's wording?

**Comment by NickvanDyke** [comment]
Ah right, I keep forgetting about server vs client 😆 thanks!

Yeah, let me think about whether `useLayoutEffect` is always correct. If not, I think you've got a good idea to not flag this when the dev has already intentionally passed an initial value to `useState` 🙂

**Comment by NickvanDyke** [comment]
No worries!

Are you on the latest version of the plugin? no-initialize-state's message should mention it.
